core/build.gradle

```diff
@@ -1,6 +1,35 @@
 apply plugin: 'com.android.library'
 apply plugin: 'bintray-release'
 apply plugin: 'jacoco'
+apply plugin: 'com.novoda.build-properties'
+
+buildProperties {
+    cli {
+        using(project)
+    }
+    bintray {
+        def bintrayCredentials = {
+            boolean isDryRun = cli['dryRun'].or(true).boolean
+            return isDryRun ?
+                    ['bintrayRepo': 'n/a', 'bintrayUser': 'n/a', 'bintrayKey': 'n/a'] :
+                    new File("${System.getenv('BINTRAY_PROPERTIES')}")
+        }
+        using(cli).or(bintrayCredentials())
+        description = '''This should contain the following properties:
+                       | - bintrayRepo: name of the repo of the organisation to deploy the artifacts to
+                       | - bintrayUser: name of the account used to deploy the artifacts
+                       | - bintrayKey: API key of the account used to deploy the artifacts
+        '''.stripMargin()
+    }
+    publish {
+        def generateVersion = {
+            boolean isSnapshot = cli['bintraySnapshot'].or(false).boolean
+            return isSnapshot ? System.getenv('GIT_COMMIT')?:'SNAPSHOT' : version
```

Comment:
git commit or jenkins build number? 

I'm leaning towards build number as it's easier to communicate

Reply:
yeah, I had no idea tbh :D
we can pick other values [too](https://wiki.jenkins.io/display/JENKINS/Building+a+software+project)), but I am happy to go for `BUILD_NUMBER` then